src/app/workflow/version-modal/version-modal.service.ts

```diff
@@ -45,36 +45,54 @@ export class VersionModalService {
   }
 
   /**
-   * Saves the version.  This contains 4 parts:
+   * Saves the version.  This contains 3 parts:
    * 1. PUT workflowVersions
-   * 2. Refresh workflow
-   * 3. Modify test parameter files
-   * 4. Refresh workflow again
-   * TODO: Skip 2 and 3 if there's no test parameter files to modify
+   * 2. Modify test parameter files
+   * 3. Refresh workflow if workflow path has changed or if a test parameter file was added/removed
    *
    * @param {WorkflowVersion} workflowVersion
    * @param {any} originalTestParameterFilePaths
    * @param {any} newTestParameterFiles
    * @memberof VersionModalService
    */
-  saveVersion(workflowVersion: WorkflowVersion, originalTestParameterFilePaths, newTestParameterFiles, workflowMode: String) {
+  saveVersion(
+    originalVersion: WorkflowVersion,
+    workflowVersion: WorkflowVersion,
+    originalTestParameterFilePaths,
+    newTestParameterFiles,
+    workflowMode: String
+  ) {
     const message1 = 'Saving workflow version';
     const message2 = 'Modifying test parameter files';
     const workflowId = this.workflowQuery.getActive().id;
+    let toRefresh = false;
+    // Checks if the workflow path was changed
+    if (originalVersion.workflow_path !== workflowVersion.workflow_path) {
+      toRefresh = true;
```

Comment:
I'm wondering if this shouldn't be handled in the web service? The web service should know whether the workflow needs to be refreshed after an update.

Arguments against:

1. There are two API calls, so there could be two refreshes
2. Maybe there is value to have the workflow updated but not refreshed.

Reply:
Ya, I made this argument years ago.  At the time, it was number 2.

Reply:
Could be wrong, but I think at the time we were working on a hotfix so we wanted to limit changes. This is for 1.9, so I think web service could be ok. 

Two refreshes would suck though